From b0375f42a27044667082e53449e534b265d7a029 Mon Sep 17 00:00:00 2001
From: Han Xu <b45815@freescale.com>
Date: Fri, 7 Aug 2015 10:37:50 -0500
Subject: [PATCH] MLK-11336: mtd: nand: enable LPSR for GPMI NAND

The LPSR turns off the power for IOMUX when suspending so restore the
IOMUX when resuming in GPMI NAND driver.

The function was not tested yet since NAND only supported on 19x19
LPDDR board.

Signed-off-by: Han Xu <b45815@freescale.com>
---
 drivers/mtd/nand/gpmi-nand/gpmi-nand.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/mtd/nand/gpmi-nand/gpmi-nand.c b/drivers/mtd/nand/gpmi-nand/gpmi-nand.c
index 479efc7..c09a6f2 100644
--- a/drivers/mtd/nand/gpmi-nand/gpmi-nand.c
+++ b/drivers/mtd/nand/gpmi-nand/gpmi-nand.c
@@ -1909,6 +1909,7 @@ static int gpmi_pm_suspend(struct device *dev)
 	struct gpmi_nand_data *this = dev_get_drvdata(dev);
 
 	release_dma_channels(this);
+	pinctrl_pm_select_sleep_state(dev);
 	return 0;
 }
 
@@ -1917,6 +1918,8 @@ static int gpmi_pm_resume(struct device *dev)
 	struct gpmi_nand_data *this = dev_get_drvdata(dev);
 	int ret;
 
+	pinctrl_pm_select_default_state(dev);
+
 	ret = acquire_dma_channels(this);
 	if (ret < 0)
 		return ret;
-- 
1.8.0

